src/hooks/internal/useField.js

```diff
@@ -17,6 +16,16 @@ function useField(componentName, { name, disabled, optional, validate, data }) {
     unregisterField,
   } = useForm(componentName);
 
+  const fieldDataRef = useRef(data);
+  const registerDataRef = useRef({
+    registerField,
+    unregisterField,
+    name,
+    disabled,
+    optional,
+    validate,
+  });
```

Comment:
So we set all these values only once now, and this is a problem.
For example, if `disabled` is updated from `false` to `true`, the form wouldn't get this update and still try to validate the field even though it's disabled now.

Here is an example that demonstrates this:
On [master](https://basis.vercel.app/playground/?data=N4Igxg9gJgpiBcIBmBXAdmALgSwmgBAIIAOxAFAJT7AA6B%2BkaAzpvtmtjgIYA2AarxQwm%2BALzU6%2BKfjRcAtjAAi2JlwBGPGFHj4kvJjAA0k6bIU6aIS8foBfOiYBOMTCkcEyJqQB4AwnkwudhhHfGIuKCh2AHNRSwAOSwA%2BL2lvADEIRzk2Dm5%2BQWFRYHZObF4BHiEmWxT6aWlgMmB8Fi5MI3wAN15sKHaYdOwYHih8WypRJIl6hulnV3d8T1m5tIBlQLAAa3xormI4kESQOrXz-D8ACxgdtQgADxl5GCOzJRV1TShLfB51EZHACSIne%2BCiqg0WgA-L8rn0YAAZAE8M4Xc4g54KcGfKFQWGrdHeAD0vhud0eaPRaSBaGIKEwqWpUnebxeliZzP%2BakBlgAcuyrITqRCvlpim0OgA6HpVYRS97KSHfezC87EqnM7wAIQZmDwnOpeF8PGwO2KlDE01oaupsr6AyGIygnhA70sFEN6NsquZ0k1fsqDo6WJgXrWJN1mH1aAD5xJmy4OzjDU9ap9nJJmWympJ-jQgWCjjRadVIEMIAM0ZiTAQoCYYGcMGYCAA2q3K3JeDxywAWAAM-YAukOfUA) branch:
1. Click the checkbox
2. Click the button
You'll see that the field is **not** validated. This is the expected behavior.

But, on [your branch](https://basis-git-fork-strahinjaajvaz-fix-formsnotrende-ea076a.vercel.app/playground/?data=N4Igxg9gJgpiBcIBmBXAdmALgSwmgBAIIAOxAFAJT7AA6B%2BkaAzpvtmtjgIYA2AarxQwm%2BALzU6%2BKfjRcAtjAAi2JlwBGPGFHj4kvJjAA0k6bIU6aIS8foBfOiYBOMTCkcEyJqQB4AwnkwudhhHfGIuKCh2AHNRSwAOSwA%2BL2lvADEIRzk2Dm5%2BQWFRYHZObF4BHiEmWxT6aWlgMmB8Fi5MI3wAN15sKHaYdOwYHih8WypRJIl6hulnV3d8T1m5tIBlQLAAa3xormI4kESQOrXz-D8ACxgdtQgADxl5GCOzJRV1TShLfB51EZHACSIne%2BCiqg0WgA-L8rn0YAAZAE8M4Xc4g54KcGfKFQWGrdHeAD0vhud0eaPRaSBaGIKEwqWpUnebxeliZzP%2BakBlgAcuyrITqRCvlpim0OgA6HpVYRS97KSHfezC87EqnM7wAIQZmDwnOpeF8PGwO2KlDE01oaupsr6AyGIygnhA70sFEN6NsquZ0k1fsqDo6WJgXrWJN1mH1aAD5xJmy4OzjDU9ap9nJJmWympJ-jQgWCjjRadVIEMIAM0ZiTAQoCYYGcMGYCAA2q3K3JeDxywAWAAM-YAukOfUA), the disabled field is validated because it keeps the old `field.disabled = false` value.